File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.js
function solution(n) {
    let answer = 0;
    let a = "";
    
    let ans = n.toString();
    let list = ans.split("");
    // n을 문자열로 변환하고, 배열로 분리한다
    
    list.sort((a, b) => b - a);
    // 배열을 내림차순으로 정렬한다
    
    for (let i = 0; i < list.length; i++) {
        a += list[i];
    } // 배열을 다시 문자열로 합친다
    
    answer = parseInt(a);
    // 문자열을 정수형으로 변환한다
    
    return answer;
}

File: 프로그래머스/lv1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public long solution(long n) {
        long answer = 0;
		String a = "";
        
		String ans = String.valueOf(n);
        String[] list = ans.split("");
	    // n을 문자, 배열로 순서대로 변환시킨다
        
        Arrays.sort(list, Collections.reverseOrder());
        // 배열을 내림차순으로 정렬한다
        
		for(int i = 0; i < list.length; i++) {
			a += list[i];
		} // 배열을 문자로 변환한다
        
		answer = Long.parseLong(a);
        // 문자를 long 타입으로 변환한다
        
        return answer;
    }
}